src/TokenInterface.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace Dhii\Parser\Tokenizer;
+
+use Dhii\Data\Container\ContainerInterface;
+
+/**
+ * Represents a token.
+ *
+ * A token is a piece of data at a specific location. It has domain-specific
+ * data attached to it.
+ *
+ * @since [*next-version*]
+ */
+interface TokenInterface extends
+        ContainerInterface,
+        LineNumberAwareInterface,
+        ColumnNumberAwareInterface
```

Comment:
Perhaps it's also worth having a token be aware of some kind of key for the resource it came from, be it a resource handle or key, file path, URL, etc.

What do you think?

Reply:
Sounds a bit extra, IMHO. Maybe at a higher level the parsing is done on a stream. In that case, the higher level can provide the stream identifier. Parsing works on just "data".